src/engine/source/logicexpr/interface/logicexpr/tokenizer.hpp

```diff
@@ -0,0 +1,235 @@
+                                       res.index());
+        }
+        else
+        {
+            return parsec::makeError<Token>("Unexpected token", pos);
+        }
+    }
+
+    /**
+     * @brief Parses an operator token from a string.
+     *
+     * @param sv The string to parse.
+     * @param pos The starting position of the string to parse.
+     * @return A parsec::Result object containing the parsed token or an error message.
+     */
+    parsec::Result<Token> operatorParser(std::string_view sv, size_t pos) const
+    {
+        if (sv.substr(pos, opstr::OR.size()) == opstr::OR)
+        {
+            return parsec::makeSuccess(Token {OrToken::create(opstr::OR, pos)}, pos + opstr::OR.size());
+        }
+        else if (sv.substr(pos, opstr::AND.size()) == opstr::AND)
+        {
+            return parsec::makeSuccess(Token {AndToken::create(opstr::AND, pos)}, pos + opstr::AND.size());
+        }
+        else if (sv.substr(pos, opstr::NOT.size()) == opstr::NOT)
+        {
+            return parsec::makeSuccess(Token {NotToken::create(opstr::NOT, pos)}, pos + opstr::NOT.size());
+        }
+        else if (sv.substr(pos, opstr::P_OPEN.size()) == opstr::P_OPEN)
+        {
+            return parsec::makeSuccess(Token {ParenthOpenToken::create(opstr::P_OPEN, pos)},
+                                       pos + opstr::P_OPEN.size());
+        }
+        else if (sv.substr(pos, opstr::P_CLOSE.size()) == opstr::P_CLOSE)
+        {
+            return parsec::makeSuccess(Token {ParenthCloseToken::create(opstr::P_CLOSE, pos)},
+                                       pos + opstr::P_CLOSE.size());
+        }
+        else
+        {
+            return parsec::makeError<Token>("Unexpected token", pos);
+        }
+    }
+
+    /**
+     * @brief Parses a token from a string by trying to parse a term token first, and if that fails, parsing an operator
+     * token.
+     *
+     * @param sv The string to parse.
+     * @param pos The starting position of the string to parse.
+     * @return A parsec::Result object containing the parsed token or an error message.
+     */
+    parsec::Result<Token> tokenParser(std::string_view sv, size_t pos) const
+    {
+        // The term parser must be tried first, to avoid scaping other tokens
+        // Term parser no should be success if the token is a operator
+        auto res = termParser(sv, pos);
+        if (res.success())
+        {
+            return res;
+        }
+        else
+        {
+            return operatorParser(sv, pos);
+        }
+    }
+
+    /**
+     * @brief Parses a queue of tokens from a string.
+     *
+     * @param sv The string to parse.
+     * @param pos The starting position of the string to parse.
+     * @return A parsec::Result object containing the parsed queue of tokens or an error message.
+     */
+    parsec::Result<std::queue<Token>> parser(std::string_view sv, size_t pos) const
```

Comment:
nitpick - desing: You could use std::find_if to find the start of each token (ignoring spaces), and then apply the token parsing logic. This eliminates the need for do-while.

```
    auto isNotSpace = [](char c) { return !std::isspace(c); };
    auto tokenEnd = sv.end();
    
    while (pos < sv.size())
    {
        auto tokenStart = std::find_if(sv.begin() + pos, sv.end(), isNotSpace);
        
        if (tokenStart == tokenEnd)
        {
            break;
        }
        // Rest of the logic
 }
```